tests/integration/test_storage_postgresql/test.py

```diff
@@ -164,6 +166,47 @@ def test_non_default_scema(started_cluster):
     assert(result == expected)
 
 
+def test_concurrent_queries(started_cluster):
+    conn = get_postgres_conn(True)
+    cursor = conn.cursor()
+    node1.query('''
+        CREATE TABLE test_table (key UInt32, value UInt32)
+        ENGINE = PostgreSQL('postgres1:5432', 'clickhouse', 'test_table', 'postgres', 'mysecretpassword')''')
+
+    cursor.execute('CREATE TABLE test_table (key integer, value integer)')
+
+    def node_insert(_):
+        for i in range(20):
+            result = node1.query("INSERT INTO test_table SELECT number, number FROM numbers(1000)", user='default')
+
+    busy_pool = Pool(10)
+    p = busy_pool.map_async(node_insert, range(10))
+    p.wait()
+    result = node1.query("SELECT count() FROM test_table", user='default')
+    print(result)
+    assert(int(result) == 20 * 10 * 1000)
```

Comment:
Does this test check somewhere how many connections we made? Maybe we can add something with 'pg_backend_pid()', it should be the same for each connection. We can insert it, and then select how many distinct values we inserted, and check that it's equal to pool size. In python we'll have to use much larger pool so that the postgres pool is saturated.

Reply:
We could limit `max_connections` in Postgres and have the pool of same size, and then have much more connections in ClickHouse. This way we'll notice if the pool is not used.

Reply:
I also had an idea to make a function similar to `contains_in_log` in helpers/cluster.py, but `count_in_log` (uses grep). And when a new connection is opened just to add some LOG_DEBUG message that a new connection was opened to host:port.

Reply:
> I also had an idea to make a function similar to `contains_in_log(message)` in helpers/cluster.py, but `count_in_log(message)` (uses grep). And when a new connection is opened just to add some LOG_DEBUG message that a new connection was opened to host:port.

Yep, counting the connections in postgres logs is also an option.

Reply:
By the way, what exactly is this test supposed to check? If it's just the fact that we can connect and run queries concurrently, maybe the above test with log is enough? It also performs queries concurrently.

Reply:
I caught some races in libpq, which are now fixed, they did not reproduce with a less hard test.
But now probably I can delete this test..

Reply:
I mean, maybe we can just add the check for the number of connections to it? Well, it's not important, they just didn't look very different so I thought maybe we can merge. If this test was useful to you, we can keep it.

Reply:
It is not possible to add a check for the number of connections to this long test, because in this test there are much more concurrent attempts to connect than the limit of the pool. So I decided to make a separate small test.